src/context.cpp

```diff
@@ -596,61 +699,172 @@ namespace librealsense
             if ( profiles.size() > 0 )
             {
                 _dev->open( realdds_profiles );
+
             }
 
             software_sensor::open( profiles );
         }
 
+        void handle_video_data( realdds::topics::device::image && dds_frame,
+                                std::shared_ptr< stream_profile_interface > prof, std::string name )
+        {
+            rs2_software_video_frame rs2_frame = {};
+
+            // prof parameter holds the real data, rs2_software_video_frame forces us to hold a pointer to it.
+            // Because we use syncer, not calling on_video_frame in the lifetime of this function, we need a shared_ptr
+            // that the syncer will hold till using the frame.
+            std::shared_ptr< rs2_stream_profile > prof_holder = std::make_shared< rs2_stream_profile >();
+            prof_holder->profile = prof.get();
+            rs2_frame.profile = prof_holder.get();
+
+            //Copying from dds into LibRS space, same as copy from USB backend.
+            //TODO - use memory pool or some other frame allocator
+            rs2_frame.pixels = new uint8_t[dds_frame.size];
+            if( !rs2_frame.pixels )
+                throw std::runtime_error( "Could not allocate memory for new frame" );
+            memcpy( rs2_frame.pixels, dds_frame.raw_data.data(), dds_frame.size );
+            // The way to release allocated memory
+            rs2_frame.deleter = []( void * ptr ) { delete[] ptr; };
+            // Info sent with image topic
+            rs2_frame.stride = dds_frame.height > 0 ? dds_frame.size / dds_frame.height : dds_frame.size;
+            rs2_frame.bpp = dds_frame.width > 0 ? rs2_frame.stride / dds_frame.width : rs2_frame.stride;
+            rs2_frame.frame_number = std::stoi( dds_frame.frame_id );
+
+            if( _md_enabled )
+                _stream_name_to_video_syncer[name].enqueue_frame( std::move( rs2_frame ), prof_holder );
+            else
+                on_video_frame( rs2_frame );
+        }
+
+        void add_video_frame_metadata( rs2_software_video_frame & frame, const realdds::topics::flexible_msg & dds_md )
+        {
+            // Always expected metadata
+            frame.timestamp = rsutils::json::get< rs2_time_t >( dds_md.json_data(), "timestamp" );
+            frame.domain = rsutils::json::get< rs2_timestamp_domain >( dds_md.json_data(), "timestamp-domain" );
+            // Expected metadata for all depth images
+            if( rsutils::json::has( dds_md.json_data(), "depth-units" ) )
+                frame.depth_units = rsutils::json::get< float >( dds_md.json_data(), "depth-units" );
+            // Other metadata fields
+            for( size_t i = 0; i < static_cast< size_t >( RS2_FRAME_METADATA_COUNT ); ++i )
+            {
+                const char * str = rs2_frame_metadata_to_string( static_cast< rs2_frame_metadata_value >( i ) );
+                if( rsutils::json::has( dds_md.json_data(), str ) )
```

Comment:
This means that if a metadata field is present that we do not know, we'll be ignoring it and no message will be given... right?

Reply:
Yes. If there is data that is unsupported by LibRS it is ignored and no message will be given about it.

Reply:
I think it might be worth a mention. We can say this is a "feature" and not a byproduct, as it makes the server more flexible.

Reply:
Added a comment